capabilityreceiver/cmd/migratevalidations/main_integration_test.go

```diff
@@ -0,0 +1,272 @@
+			updated_at
+		)
+		VALUES ($1, $2, $3, $4, $5, $6, $7)`
+
+	getFHIREndpointInfoStatement := `
+		SELECT updated_at, validation_result_id
+		FROM fhir_endpoints_info_history
+		WHERE url=$1
+	`
+
+	getValidationResultStatement := `
+		SELECT COUNT(*)
+		FROM validation_results
+		WHERE id=$1
+	`
+
+	getValidationStatement := `
+		SELECT COUNT(*)
+		FROM validations
+		WHERE validation_result_id=$1
+	`
+
+	// Add metadata
+	metadataID1, err := store.AddFHIREndpointMetadata(ctx, &testMetadata1)
+	th.Assert(t, err == nil, fmt.Sprintf("Error while adding first metadata object: %s", err))
+	metadataID2, err := store.AddFHIREndpointMetadata(ctx, &testMetadata2)
+	th.Assert(t, err == nil, fmt.Sprintf("Error while adding first metadata object: %s", err))
+
+	// Put two FHIR endpoints in the history table
+	tlsVersion := "1.3"
+	mimeTypes := []string{"application/json+fhir"}
+	firstTime := time.Now().UTC().Round(time.Microsecond)
+	url1 := "www.testurl.com/cerner/DSTU2"
+	_, err = store.DB.ExecContext(ctx, addFHIREndpointInfoStatement, url1, "I", capStat1, tlsVersion, pq.Array(mimeTypes), metadataID1, firstTime)
+	th.Assert(t, err == nil, fmt.Sprintf("Error when adding to the database %s", err))
+
+	secondTime := time.Now().UTC().Round(time.Microsecond)
+	url2 := "www.testurl.com/epic/DSTU2"
+	_, err = store.DB.ExecContext(ctx, addFHIREndpointInfoStatement, url2, "I", capStat2, tlsVersion, pq.Array(mimeTypes), metadataID2, secondTime)
+	th.Assert(t, err == nil, fmt.Sprintf("Error when adding to the database again %s", err))
+
+	resultCh := make(chan Result)
+
+	// Check that data only updates the first URL
+	defaultArgs := make(map[string]interface{})
+	defaultArgs["historyArgs"] = historyArgs{
+		fhirURL:   url1,
+		store:     store,
+		result:    resultCh,
+		isHistory: true,
+	}
+
+	go addToValidationTable(ctx, &defaultArgs)
+	for res := range resultCh {
+		th.Assert(t, res.URL == url1, fmt.Sprintf("Returned result URL is not equal to %s, is instead %s", url1, res.URL))
+		close(resultCh)
+	}
+
+	historyRows, err := store.DB.QueryContext(ctx, getFHIREndpointInfoStatement, url1)
+	th.Assert(t, err == nil, fmt.Sprintf("error getting data from fhir_endpoints_info: %s", err))
+	// Check to make sure exactly 1 ID was updated
+	count := 0
+	var valID int
+	for historyRows.Next() {
+		th.Assert(t, count < 1, fmt.Sprintf("should only be one item in the database for this URL"))
+		var receivedTime time.Time
+		err = historyRows.Scan(&receivedTime, &valID)
+		th.Assert(t, err == nil, fmt.Sprintf("Error while scanning the rows of the history table for URL %s. Error: %s", url1, err))
+		th.Assert(t, receivedTime.Equal(firstTime), fmt.Sprintf("The time was updated to %+v from %+v", receivedTime, firstTime))
+		count++
+	}
+	th.Assert(t, count == 1, "should be one item in the database, instead is 0")
+
+	// Make sure that ID was added to the validation results table
+	valResRows, err := store.DB.QueryContext(ctx, getValidationResultStatement, valID)
```

Comment:
Here you could use probably QueryRowContext to directly scan the count rather than having to call valResRows.next(), since this query should only return one row anyways